pallets/light-proposals/src/lib.rs

```diff
@@ -0,0 +1,257 @@
+	FunctionSignature, Proposal, ProposalHandlerTrait, ProposalHeader, ProposalKind, ResourceId,
+};
+
+use frame_support::{pallet_prelude::DispatchError, traits::Get};
+pub use pallet::*;
+use scale_info::TypeInfo;
+use sp_std::{convert::TryInto, prelude::*};
+
+use webb_proposals::{evm::AnchorUpdateProposal, Nonce, TypedChainId};
+
+/// Function sig for anchor update
+pub const ANCHOR_UPDATE_FUNCTION_SIGNATURE: [u8; 4] = [0x26, 0x57, 0x88, 0x01];
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+
+	use dkg_runtime_primitives::ProposalHandlerTrait;
+
+	use frame_support::{
+		dispatch::{fmt::Debug, DispatchResultWithPostInfo},
+		pallet_prelude::{ValueQuery, *},
+	};
+	use frame_system::pallet_prelude::*;
+	use webb_light_client_primitives::{
+		traits::{LightClientHandler, StorageProofVerifier},
+		types::LightProposalInput,
+	};
+
+	pub type LightProposalInputOf<T> = LightProposalInput<<T as Config>::MaxProofSize>;
+
+	#[pallet::pallet]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	/// The module configuration trait.
+	pub trait Config: frame_system::Config + pallet_bridge_registry::Config {
+		/// The overarching event type.
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		/// Light client interface for the pallet
+		type LightClient: LightClientHandler;
+
+		/// Storage proof verifier for the pallt
+		type StorageProofVerifier: StorageProofVerifier;
+
+		/// Proposer handler trait
+		type ProposalHandler: ProposalHandlerTrait<
+			MaxProposalLength = <Self as pallet::Config>::MaxProposalLength,
+		>;
+
+		/// Max length of submitted proof
+		#[pallet::constant]
+		type MaxProofSize: Get<u32> + TypeInfo + Clone + Debug + PartialEq + Eq;
+
+		/// Max length of a proposal
+		#[pallet::constant]
+		type MaxProposalLength: Get<u32>
+			+ Debug
+			+ Clone
+			+ Eq
+			+ PartialEq
+			+ PartialOrd
+			+ Ord
+			+ TypeInfo;
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		ProposalSubmitted { typed_chain_id: TypedChainId, proposal: LightProposalInputOf<T> },
+	}
+
+	#[pallet::storage]
+	#[pallet::getter(fn resource_id_to_nonce)]
+	/// Mapping of resource to nonce
+	pub type ResourceIdToNonce<T: Config> = StorageMap<_, Blake2_256, ResourceId, u32, ValueQuery>;
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Cannot fetch bridge details
+		CannotFetchBridgeDetails,
+		/// Cannot fetch bridge metadata
+		CannotFetchBridgeMetadata,
+		/// Proof verification failed
+		ProofVerificationFailed,
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::call_index(0)]
+		#[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().reads_writes(1,1))]
+		pub fn submit_proposal(
+			origin: OriginFor<T>,
+			typed_chain_id: TypedChainId,
```

Comment:
```suggestion
		resource_id: ResourceId,
```

ResourceId will contain (typed chainId and Vanchor contract address )